src/client/app/components/admin/users/CreateUserModalComponent.tsx

```diff
@@ -0,0 +1,234 @@
+	};
+
+	// Handlers for each type of input change
+	const handleStringChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			[e.target.name]: e.target.value
+		}));
+	};
+
+	const handleRoleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			role: e.target.value as UserRole
+		}));
+	};
+
+	// Methods to reset form fields
+	const resetForm = () => {
+		setUserDetails(userDefaults);
+	};
+
+	const resetPasswordFields = () => {
+		setUserDetails(prevDetails => ({
+			...prevDetails,
+			password: '',
+			confirmPassword: ''
+		}));
+	};
+	// End of reset form methods
+
+	// Modal show/close
+	const [showModal, setShowModal] = useState(false);
+
+	const handleShowModal = () => setShowModal(true);
+
+	const handleCloseModal = () => {
+		resetPasswordFields();
+		setShowModal(false);
+	};
+	// End Modal show/close
+
+	const handleSubmit = async () => {
+		const newUser: User = { username: userDetails.username, role: userDetails.role, password: userDetails.password, note: userDetails.note };
+		createUser(newUser)
+			.unwrap()
+			.then(() => {
+				showSuccessNotification(translate('users.successfully.create.user') + userDetails.username);
+				resetForm();
+				handleCloseModal();
+			})
+			.catch(error => {
+				showErrorNotification(translate('users.failed.to.create.user') + userDetails.username + ' ' + error.data.message);
+				resetPasswordFields();
+			});
+	};
+
+	const tooltipStyle = {
+		...tooltipBaseStyle,
+		// Only an admin can create a meter.
+		tooltipUsersView: 'help.admin.users'
+	};
+
+	return (
+		<>
+			<Button color='secondary' onClick={handleShowModal}>
+				{translate('create.user')}
+			</Button>
+			<Modal isOpen={showModal} toggle={handleCloseModal} size='lg'>
+				<ModalHeader>
+					{translate('create.user')}
+					<TooltipHelpComponent page='users-create' />
+					<div style={tooltipStyle}>
+						<TooltipMarkerComponent page='users-create'
+							helpTextId={tooltipStyle.tooltipUsersView}
+						/>
+					</div>
+				</ModalHeader>
+				<ModalBody>
+					<Container>
+						<Row xs='1' lg='2'>
+							<Col>
+								<FormGroup>
+									<Label for='username'>
+										{translate('username')}
+									</Label>
+									<Input
+										id='username'
+										name='username'
+										type='text'
+										value={userDetails.username}
+										onChange={e => handleStringChange(e)}
+										invalid={!userDetails.username}
+									/>
+								</FormGroup>
+							</Col>
+							<Col>
+								<FormGroup>
+									<Label for='role'>
+										{translate('user.role')}
+									</Label>
+									<Input
+										id='role'
+										name='role'
+										type='select'
+										value={userDetails.role}
+										onChange={handleRoleChange}
+										invalid={userDetails.role === UserRole['user.role.select']}
+									>
+										{Object.entries(UserRole).map(([role, value]) => (
```

Comment:
This is done differently than the meter modal where the need to select is gone once you make a selection (cannot choose it again). I think that is a little nicer. Please either fix of put in a TODO.

Reply:
OK, this now works in a similar manner.